File: game-frame-server/game-network-message/src/main/java/com/xinyue/network/message/impl/GateMessageRequest.java
package com.xinyue.network.message.impl;
import com.xinyue.network.message.common.AbstractGameMessage;
import com.xinyue.network.message.common.GameMessageMetaData;
import com.xinyue.network.message.common.GameMessageType;
import com.xinyue.utils.JsonUtil;
import com.xinyue.network.EnumServerType;

//河南心悦网络科技有限公司   王广帅
//连接认证

@GameMessageMetaData(serverType = EnumServerType.GATE,messageId = 1001, messageType =GameMessageType.REQUEST)
public class GateMessageRequest extends AbstractGameMessage {
	//消息内容
	private String message;
	
	public GateMessageRequest(){
	}
	public GateMessageRequest(String message){
		this.message = message;		
	}
	
	
	public GateMessageResponse newResponse() {
		GateMessageResponse response = new GateMessageResponse();
		copyMessageHead(response.getMessageHead());
		return response;
	}
	
	public void setMessage (String message){
		this.message = message;
	}
		
	public String getMessage(){
		return message;
	}
	
	@Override
	public void decodeBody(byte[] bytes) throws Exception {
		com.xinyue.network.message.impl.proto.Gate.GateMessageRequestModel body = com.xinyue.network.message.impl.proto.Gate.GateMessageRequestModel.parseFrom(bytes);
		
			this.message = body.getMessage();
	}
	
	@Override
	public byte[] encodeBody() {  
		 if(this.getMessageHead().getErrorCode() != 0){
		  	return null;
		 }
		 
		 com.xinyue.network.message.impl.proto.Gate.GateMessageRequestModel.Builder builder = com.xinyue.network.message.impl.proto.Gate.GateMessageRequestModel.newBuilder();
		  builder.setMessage(this.message);
		 return builder.build().toByteArray();
	}
	@Override
	public String toString(){
		String info = "GateMessageRequest->" + JsonUtil.objToJson(this);
		return info;
	}
}

File: game-frame-client/xinyue-game-client/Assets/Scripts/game-network/game-message-impl/GateMessageRequest.cs
using Assets.Scripts.game_network.game_message;
using ProtoBuf;
using System.IO;
using com.xinyue.network.message.impl.proto;

//河南心悦网络科技有限公司   王广帅
namespace Assets.Scripts.game_network.game_message_impl
{
	[GameMessageMeta(1001,EnumMessageType.REQUEST,EnumServerType.GATE)]
	public class GateMessageRequest : AbstractGameMessage
	{
	
		// 消息内容
		public string message;
		
		public GateMessageRequest()
		{
		
		}
		public GateMessageRequest(string message){
			this.message = message;		
		}
		
		public override byte[] EncodeBody()
		{
			
			
			GateMessageRequestModel entity = new GateMessageRequestModel();
			
			entity.message = this.message;
			 MemoryStream ms = new MemoryStream();
			 using(ms){
             	Serializer.Serialize(ms, entity);
             	return ms.ToArray();
             }
		}
	
		public override void DecodeBody(byte[] bytes)
		{
			
			
			MemoryStream ms = new MemoryStream(bytes);
			using(ms){
				GateMessageRequestModel entity = Serializer.Deserialize<GateMessageRequestModel>(ms);
							this.message = entity.message;
			}
		}
	}
}
